mongodb/context_strategy.go

```diff
@@ -0,0 +1,144 @@
+package mongodb
+
+import (
+	"context"
+	"time"
+)
+
+// ContextStrategy is an interface that represents strategy for providing contexts for running MongoDB requests
+type ContextStrategy interface {
+	// Append is the context used for EventStore.Append() calls
+	Append() (context.Context, context.CancelFunc)
+	// GetEventsFor is the context used for EventStore.GetEventsFor() calls
```

Comment:
In go a comment should not be needed on the interface methods because the interface express already the actions that are expressing the methods. In addition having a such big interface create constraints when used in other packages or for testing purposes.

Reply:
Soudns good, will change descriptions